html/includes/graphs/device/nfsen_channel_common.inc.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+$simple_rrd = true;
+
+if (!is_array($config['nfsen_rrds'])) {
+    $config['nfsen_rrds'] = array($config['nfsen_rrds']);
+}
+
+foreach ($config['nfsen_rrds'] as $nfsenrrds) {
+    if ($nfsenrrds[(strlen($nfsenrrds) - 1)] != '/') {
+        $nfsenrrds .= '/';
+    }
+
+    $nfsenHostname=$device['hostname'];
+    if ($config['nfsen_split_char']) {
+        $nfsenHostname=str_replace('.', $config['nfsen_split_char'], $device['hostname']);
+    }
+    $rrd_filename=$nfsenrrds.$nfsenHostname.'/'.$vars['channel'].'.rrd';
+
+    if (is_file($rrd_filename)) {
+        $flowtypes = array('tcp', 'udp', 'icmp', 'other');
+
+        $rrd_list   = array();
+        $nfsen_iter = 1;
+        foreach ($flowtypes as $flowtype) {
+            $rrd_list[$nfsen_iter]['filename']  = $rrd_filename;
```

Comment:
Rather than having to keep track of $nfsen_iter, you can just set a complete array once:
```
$rrd_list[] = array(
    'filename'  => $rrd_filename,
    'descr' => $flowtype,
    'ds' => $dsprefix.$flowtype
);
```